src/lib/radio/radio.ts

```diff
@@ -507,7 +510,7 @@ export class MatRadioButton extends _MatRadioButtonMixinBase
     this._removeUniqueSelectionListener =
       _radioDispatcher.listen((id: string, name: string) => {
         if (id != this.id && name == this.name) {
-          this.checked = false;
+          this.checked = this._inputElement.nativeElement.checked;
```

Comment:
Might want to add a comment that explains the issue?

Reply:
Done. Thanks for review